restic/internal/restore.py

```diff
@@ -12,7 +12,8 @@ def run(restic_base_command,
         cmd.extend(['--include', include])
 
     if exclude:
-        cmd.extend(['--exclude', exclude])
+        for exclude_path in exclude:
```

Comment:
There's a bit of a dangerous gotcha here for clients that depended on the previous semantics.

Can we  add a check here for if `exclude` is a string and then throw an Exception in that case?

Because otherwise, if a consuming app was calling `restic.restore(exclude='/foo/bar')`, we're going to turn it into `--exclude / --exclude f --exclude o...`

We could theoretically check if it's a string vs. a list and handle both, but I'd rather just fail hard on strings to reduce complexity.

Reply:
Good catch, I thought about it while writing this as I am actually actually using the lib in this configuration
Your proposal is fair and I implemented it in a separated commit. Thanks !

Reply:
Thanks for those edits!

I made a few small changes:

- I added a more specific error for this case, as I generally prefer module-level exceptions for things that are scoped to a particular module.
- I got rid of the mock execute in the test, since it doesn't actually get called and I'd rather keep it simple.
- I added back the test for a single exclude path using the new semantics